astropy/io/fits/hdu/image.py

```diff
@@ -638,7 +646,64 @@ def _writedata_internal(self, fileobj):
 
             size += output.size * output.itemsize
 
-        return size
+            return size
+
+    @staticmethod
+    def _dask_byteswap(array):
+        import dask.array
+        from dask import delayed
+
+        @delayed
```

Comment:
This implementation might not be the most efficient, see https://github.com/dask/dask/issues/5689